【Task18】Implement autoplay and pagination for slideshow

csakyo opened the pull request:
#[Task18]
スライドショーにドットのページネーションを作りましょう
https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#18

[仕様］
・それぞれのドットではクリッカブルになっていて、押下するとその画像に切り替わる
・それとともに1/5も切り替わる
・3秒毎に次のスライドに自動で切り替わるauto機能

## [codesandbox](https://codesandbox.io/s/cool-mestorf-21gqtz?file=/main.js) / Latest commit:https://github.com/csakyo/homework/pull/32/commits/b18b4269a644382f0824c764450937a605f5bc47
## [Comment for this Task]
Please let me know if there are any appropriate argument or function names. 
If you have a more concise way of writing, please let me know.

csakyo committed:
feat: Implement autoplay and pagination for slideshow

csakyo committed:
fix: Correction of argument names

csakyo commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+      setNumberOfPage(imgData);
+    })
+  } 
+}
```
These functions have been added since the last issue.

ghost commented:
IMO
If the button is pressed when the auto function is running, it will move to the next slide in a short interval depending on the timing.
I thought it would be easier for users to view the content if the button was pressed and the slide moved 3 seconds after the moment it was pressed.
https://user-images.githubusercontent.com/43788628/164957347-237ea06d-87cd-4016-a432-ba94ec01251e.mov

kenmori commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
```
👍

kenmori commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+
+const renderPagenation = (imgData) => {
+  const pagenation = createElementWithClassName("ul", "pagenation");
+  pagenation.id = 'pagenation';
```
Q
`pagenation.id`というのは何に使っているものですか？

csakyo replied:
Deleted as not in use!Thank you for pointing out it!
fixed commit : https://github.com/csakyo/homework/pull/32/commits/3db3bb99f7af10d25d8e37e57ed870355a5e67ac

kenmori commented on question18/.gitignore:
```diff
@@ -0,0 +1,2 @@
+.DS_Store
+img
```
なんでimgなのか... 🤔

csakyo replied:
以前の課題で画像も一緒にgitにあげてしまったのですが、レビューする際に邪魔だなと感じたので、imgディレクトリはignoreしてます:eyes:

kenmori commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+}
+
+
+const getImgListsFragment = (imgData) => {
```
戻り値の変数名と関数名を合わせる
```suggestion
const getFragmentImglists = (imgData) => {
```

csakyo replied:
Thank you for your review!I fixed it!
fixed commit : https://github.com/csakyo/homework/pull/32/commits/50243231ef5f0a7846c635cae7867c5df7e073a5

csakyo committed:
feat: Add autoplay reset function

csakyo committed:
Refactor: Commonize similar functions

csakyo committed:
Changed imgNum declaration to use object

csakyo committed:
Refactor: Removed unnecessary id settings.

csakyo committed:
Refactor: Rename the function to an appropriate name

csakyo commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+    switchDisableForBtn(imgData);
+    setNumberOfPage(imgData);
+  });
+}
```
Thank you for your ideas!! :sob: I'm so glad I didn't realize you could put it together like this! I have taken your ideas into consideration. I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/32/commits/187186d270fe31b91577974a3f18efd6831350dc

csakyo commented:
> IMO If the button is pressed when the auto function is running, it will move to the next slide in a short interval depending on the timing. I thought it would be easier for users to view the content if the button was pressed and the slide moved 3 seconds after the moment it was pressed. https://user-images.githubusercontent.com/43788628/164957347-237ea06d-87cd-4016-a432-ba94ec01251e.mov

Thank you for your review!! That's right! Fixed so that the timer resets when the button is pressed. Please confirm.
fixed commit : https://github.com/csakyo/homework/pull/32/commits/d777233e235a017d856d9762040eea8014be3ca1

csakyo committed:
Refactor: Deleted unwanted lines

csakyo commented on question18/main.js:
```diff
@@ -0,0 +1,191 @@
+};
+
+
+let imgNum = 0;
```
Thank you for your review! I learned a lot because I can't come up with it on my own. I thought it would be better to use this one because the use of const prevents unexpected reassignment. I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/32/commits/16b86688e662a216b2c0997bc8f24038cded15ca
